apps/labrinth/src/routes/v3/users.rs

```diff
@@ -244,27 +245,21 @@ pub async fn collections_list(
     let id_option = DBUser::get(&info.into_inner().0, &**pool, &redis).await?;
 
     if let Some(id) = id_option.map(|x| x.id) {
-        let user_id: UserId = id.into();
-
-        let can_view_private =
-            user.is_some_and(|y| y.role.is_mod() || y.id == user_id);
-
-        let project_data = DBUser::get_collections(id, &**pool).await?;
+        let collection_data = DBUser::get_collections(id, &**pool).await?;
 
         let response: Vec<_> = crate::database::models::DBCollection::get_many(
-            &project_data,
+            &collection_data,
             &**pool,
             &redis,
         )
         .await?
         .into_iter()
-        .filter(|x| {
-            can_view_private || matches!(x.status, CollectionStatus::Listed)
-        })
-        .map(Collection::from)
+        .filter(|x| matches!(x.status, CollectionStatus::Listed))
```

Comment:
This `filter` call is now redundant, as the `filter_visible_collections` call below already accounts for collection visibility based on collection status, user role, and user ownership.

Additionally, removing this call resolves a CI failure in this PR caused by the `collections_scopes` test in Labrinth no longer passing. That test asserts that private collections are visible to their owners, and that's no longer the case with this redundant `filter` call.

Reply:
I thought it might still be necessary because Unlisted collections are still permitted by the `is_hidden` check in `filter_visible_collections`. Let me workshop this a bit.

Reply:
That latest PR should work. I cannot run `cargo test` on my local machine because it OOMs, but hopefully this passes.